File: backend/app/schemas/reminders.py
from __future__ import annotations

from datetime import datetime
from typing import Literal, Optional

from pydantic import BaseModel, Field


class ReminderCreate(BaseModel):
    appointment_id: str
    patient_id: str
    channel: Literal["sms", "email", "push"]
    send_at: datetime
    template_id: str
    high_risk: bool = Field(default=False)
    additional_context: Optional[str] = None


class ReminderResponse(BaseModel):
    reminder_id: str
    status: Literal["scheduled", "sent", "error"]
    message: Optional[str] = None

File: frontend/src/api/reminders.ts
import { apiClient } from "./client";

export interface ReminderPayload {
  appointment_id: string;
  patient_id: string;
  channel: "sms" | "email" | "push";
  send_at: string;
  template_id: string;
  high_risk?: boolean;
  additional_context?: string;
}

export interface ReminderResponse {
  reminder_id: string;
  status: "scheduled" | "sent" | "error";
  message?: string;
}

export const createReminder = async (payload: ReminderPayload) => {
  const { data } = await apiClient.post<ReminderResponse>("/reminders", payload);
  return data;
};
